test/test_serialization.h

```diff
@@ -271,37 +271,101 @@ TEST(serialization, serialize_dropout) {
   check_sequential_network_model_serialization(net);
 }
 
-TEST(serialization, serialize_recurrent_cell) {
+TEST(serialization, serialize_gru_cell) {
+  network<sequential> net;
+
+  std::string json = R"(
+        {
+            "nodes": [
+                {
+                    "type": "gru_cell",
+                    "in_size": 100,
+                    "out_size": 20,
+                    "has_bias": true
+                }
+            ]
+        }
+        )";
+
+  net.from_json(json);
+
+  EXPECT_EQ(net[0]->layer_type(), "gru-cell");
```

Comment:
We should really standardize whether we are using underscores or hyphens. Layer types give hyphens and serialization uses underscores. But it is not a topic of this PR, as it exists in all layers.
/cc: @edgarriba

Reply:
Actually I've had a lot of merge conflicts because of this in both ways (from - to _ and vice-versa)

Reply:
hyphens for string looks better to me. But open to strongest opinions

Reply:
In my opnion, layer type strings should be unified.

And while it's OK to have this virtual method,
```cpp
virtual std::string layer::layer_type() const = 0;
```

It should be possible to retrieve layer type name statically for serialization.
In that way, we don't need to use different names in `serialization_layer_list.h`.